### PR TITLE
Add TagFindCommand

### LINKED ISSUE #67: As an user, I can search for a tag

### PR DESCRIPTION
**Pull request by akgrenSoar**
Closes #67 

Add TagFindCommand changelog:
- Add TagFindCommand
- Update Userguide
- Add relevant unit tests

Other Changes:
- TagName now allow spaces
- Commands in UserGuide are now in lowercase

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -244,4 +250,4 @@ Action | Format, Examples
 **Tag Delete**| `tag delete INDEX` <br> e.g., `tag delete 1` 
 **Tag Edit**| `tag edit INDEX t/TAG` <br> e.g., `tag edit 1 t/CS2101` 
 **Tag List** | `tag list`
-**Tag View** | `tag view INDEX`<br> e.g., `tag view 3`
+**Tag Find** | `tagFind  KEYWORD [MORE_KEYWORDS]`<br> e.g., `tagfind friends colleagues`
```
Comment (siqiang-ng) [inline docs/UserGuide.md]: so is the command tagFind or tagfind?

**Review by siqiang-ng — COMMENTED** [review]
LGTM other than the above comment to clarify

File: docs/UserGuide.md (hunk 1 above)
Comment (akgrenSoar) [reply docs/UserGuide.md]: It is supposed to be `tagfind`. Pushed a commit to change all commands to follow lowercase

File: src/test/java/seedu/address/testutil/TagBuilder.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package seedu.address.testutil;
+
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.TagName;
```
Comment (yeo-kexin) [inline src/test/java/seedu/address/testutil/TagBuilder.java]: conflicts with master branch

**Review by yeo-kexin — CHANGES_REQUESTED** [review]
Resolve merge conflicts in TagBuilder.java

**Review by yeo-kexin — APPROVED** [review]
LGTM

**Pull request merged**